services/src/main/java/io/druid/cli/CliHistorical.java

```diff
@@ -68,6 +69,8 @@ public CliHistorical()
           @Override
           public void configure(Binder binder)
           {
+            // register Server before binding ZkCoordinator to ensure HTTP endpoints are available immediately
+            LifecycleModule.register(binder, Server.class);
```

Comment:
Hi @xvrl , not sure if you remember this. I am curious why this line is needed? It looks like historical's HTTP endpoints won't work without this line even though it uses a different Jetty Server instance from the one registered here. Does instantiating a random Server have some good side-effects that are necessary here?